### ISSUE TITLE
Unable to move cursor (left/right) in searchbar - Reported by @daraksha-dk

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#10182]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to any chat
2. click on emoji picker 
3. Type any keyword
4. Try pressing left/right arrow key 

## Expected Result:
User should be able to move the cursor left or right when focused

## Actual Result:
User unable to move the cursor left or right when focused

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- Desktop App

**Version Number:** 1.1.86-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
![image - 2022-08-02T003648 493](https://user-images.githubusercontent.com/43995119/182292890-79be6466-db24-4df2-bd6f-ee8105ee0b5a.png)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @daraksha-dk
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1657462486325009

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by pecanoro** [comment]
I was able to reproduce it, it works fine when typing a message but it's kind of broken in the emoji picker. I also noticed that moving up in the emoji picker is broken when the cursor goes out of sight.

**Comment by dylanexpensify** [comment]
internal: https://www.upwork.com/ab/applicants/1554740356535685120/job-details
external: https://www.upwork.com/jobs/~0126faf1bb35fa93b6

**Comment by errahsoufiane** [comment]
**### Proposal**
https://github.com/Expensify/App/blob/28778a093d00dc8b882592886145e66d242d9159/src/components/EmojiPicker/EmojiPickerMenu/index.js#L150
This happens because of this line when any arrow key is pressed. To make left, right arrows work as default behavior in input box, this line should be removed.
Also I think default behavior is more user friendly than **switching to emoji tab** when right arrow pressed, even though cursor is at the end of input text. Down arrow key is already doing this.

**Solution**
do default behavior when arrow left or right key is pressed while search input is focused
```
            if (keyBoardEvent.key.startsWith('Arrow')) {
+++         if (!this.state.isFocused || (keyBoardEvent.key !== 'ArrowLeft' && keyBoardEvent.key !== 'ArrowRight')) {
                    keyBoardEvent.preventDefault();

                    // Move the highlight when arrow keys are pressed
                    this.highlightAdjacentEmoji(keyBoardEvent.key);
+++         }
                return;
            }
```

https://user-images.githubusercontent.com/108141252/182575325-ca08386f-42cf-411f-9b60-e88f14b702c5.mov

**Comment by syedsaroshfarrukhdot** [comment]
**Proposal**
https://github.com/Expensify/App/blob/28778a093d00dc8b882592886145e66d242d9159/src/components/EmojiPicker/EmojiPickerMenu/index.js#L150

The bug occurs because of keyBoardEvent.preventDefault(); which disable keyobard key default behaviour on the input field in search box when arrow keys on keyboard are pressed.

So we could add a condition on   keyBoardEvent.preventDefault(); to avoid this behaviour that if isFocused state true then avoid prevent devaulft behaiour.

 if (!this.state.isFocused){
     keyBoardEvent.preventDefault();
  }

https://user-images.githubusercontent.com/81307212/182581072-f1d1dc60-564e-48e1-8f49-6761f8ea1740.mov

**Comment by jayeshmangwani** [comment]
#### Proposal


We will check if the search input is focused and ArrowLeft or ArrowRight pressed then return from keyDownHandler and do nothing
 
https://github.com/Expensify/App/blob/28778a093d00dc8b882592886145e66d242d9159/src/components/EmojiPicker/EmojiPickerMenu/index.js#L148
```diff
+if (this.searchInput && this.searchInput.isFocused() && _.contains(['ArrowLeft', 'ArrowRight'], keyBoardEvent.key)) {
+  return;
+}
```

https://user-images.githubusercontent.com/35371050/182659716-604b00ab-2dc4-473b-9e8f-a9f7b6e83d10.mov

**Comment by neil-marcellini** [comment]
@errahsoufiane you're right that `e.preventDefault()` seems to be breaking the arrow keys in the search bar.

That line was added in this commit https://github.com/Expensify/App/commit/e98cf8e67fbfaa53e1e523f8503700b57412a158 in this PR https://github.com/Expensify/App/pull/9473.

Here's the issue they were trying to solve https://github.com/Expensify/App/issues/9413. We need to find a solution that fixes both problems.

**Comment by errahsoufiane** [comment]
> Maybe we could `e.preventDefault()` only when arrow key is up or down. (not a proposal)

yes, that is already my proposal :)

**Comment by errahsoufiane** [comment]
> @errahsoufiane you're right that `e.preventDefault()` seems to be breaking the arrow keys in the search bar.
> 
> That line was added in this commit [e98cf8e](https://github.com/Expensify/App/commit/e98cf8e67fbfaa53e1e523f8503700b57412a158) in this PR #9473.
> 
> Here's the issue they were trying to solve #9413. We need to find a solution that fixes both problems.

yes, [my proposal](https://github.com/Expensify/App/issues/10182#issuecomment-1203713120) will fix both problems. 
`keyBoardEvent.preventDefault();` should be called only when arrow key is up or down in the case search input is focused.
If search input is not focused, arrow left, right should work between emojis

**Comment by syedsaroshfarrukhdot** [comment]
> @errahsoufiane you're right that `e.preventDefault()` seems to be breaking the arrow keys in the search bar.
> 
> That line was added in this commit [e98cf8e](https://github.com/Expensify/App/commit/e98cf8e67fbfaa53e1e523f8503700b57412a158) in this PR #9473.
> 
> Here's the issue they were trying to solve #9413. We need to find a solution that fixes both problems.

Yes, So that's why we need to check is that input this.searchInput.isFocused if it is false then only keyBoardEvent.preventDefault(); should be called and if  this.status.focused is true keyBoardEvent.preventDefault();  should not be called  it is mentioned in my proposal. This way we don't need to e.preventDefault() on arrow keys it should be on the input search field which will not lead to another regression.

**Comment by syedsaroshfarrukhdot** [comment]
> @errahsoufiane you're right that `e.preventDefault()` seems to be breaking the arrow keys in the search bar.
> 
> That line was added in this commit [e98cf8e](https://github.com/Expensify/App/commit/e98cf8e67fbfaa53e1e523f8503700b57412a158) in this PR #9473.
> 
> Here's the issue they were trying to solve #9413. We need to find a solution that fixes both problems.

Yes, The solution to both problems is to not check isFocused state and only call e.preventdefault when ArrowUp and Arrow DownKey is Pressed 

So we could do something like this.
if (keyBoardEvent.key === 'ArrowUp' || keyBoardEvent.key === 'ArrowDown'){
                keyBoardEvent.preventDefault();
}
and both problems will be solved.

**Comment by errahsoufiane** [comment]
```!this.state.isFocused || ``` **(this is OR, not AND)**
this condition doesn't regress https://github.com/Expensify/App/issues/9413
Arrow Up/Down key trigger will always call `keyBoardEvent.preventDefault();` regardless of input focused status

The key difference between my proposal and @syedsaroshfarrukhdot's proposal is:
`Also I think default behavior is more user friendly than switching highlight to emoji when right arrow pressed, even though cursor is at the end of input text. Down arrow key is already doing this.` which I already commented on **my proposal**
When input cursor is at the end of text, let's trigger Arrow Right!
My proposal - does nothing. still focused on input
@syedsaroshfarrukhdot's proposal - switch highlight to first emoji, losing input focus (this is same action as Arrow Down)

**Comment by syedsaroshfarrukhdot** [comment]
> `!this.state.isFocused || ` **(this is OR, not AND)** this condition doesn't regress #9413 Arrow Up/Down key trigger will always call `keyBoardEvent.preventDefault();` regardless of input focused status
> 
> The key difference between my proposal and @syedsaroshfarrukhdot's proposal is: `Also I think default behavior is more user friendly than switching highlight to emoji when right arrow pressed, even though cursor is at the end of input text. Down arrow key is already doing this.` which I already commented on [my proposal](https://github.com/Expensify/App/issues/10182#issuecomment-1203713120) When input cursor is at the end of text, let's trigger Arrow Right! My proposal - does nothing. still focused on input @syedsaroshfarrukhdot's proposal - switch highlight to first emoji, losing input focus (this is same action as Arrow Down)

Yes, I agree with you this is the only key difference between our proposals.

**Comment by eVoloshchak** [comment]
> As this is a regression caused by #9473, this is supposed to be handled by that PR. **But** the PR was created a month back so we will handle that through new proposals.

This was present before my PR

https://user-images.githubusercontent.com/9059945/183117211-6e43448f-f489-4f89-a558-5fd6f960d8d4.mp4

**Comment by neil-marcellini** [comment]
@eVoloshchak You might be right. Would you please link me to the commit before your PR was merged and also explain what you think is causing the problem?

**Comment by eVoloshchak** [comment]
> Would you please link me to the commit before your PR was merged?

Sure: https://github.com/Expensify/App/commit/92186e50db3d37064a07d0a5c9ddcf81df39d3ad

> And also explain what you think is causing the problem

I think @parasharrajat is right, it's caused by https://github.com/Expensify/App/pull/9473
If you remove this line, it works fine (while breaking https://github.com/Expensify/App/issues/9413 probably)
https://github.com/Expensify/App/blob/9855ad8c4abb39d9d1710a70f59f23b1dd887aef/src/components/EmojiPicker/EmojiPickerMenu/index.js#L150

**Comment by neil-marcellini** [comment]
Thank you for clarifying. Yes #9473 by @JosueEchandiaAsto caused this. @rushatgabhane why did you ask @eVoloshchak to fix it? @eVoloshchak which PR did you create and how is it related to this? Sorry for the confusion. I'm trying to straighten this out.

@rushatgabhane could you please lead an RCA for this since you were the C+ on that PR?

**Comment by eVoloshchak** [comment]
> @eVoloshchak which PR did you create and how is it related to this? 

I created https://github.com/Expensify/App/pull/9954, which fixes the issue with moving to and from the search bar with keyboard arrows

> Sorry for the confusion. I'm trying to straighten this out.

No problem, I had to check this myself too

**Comment by neil-marcellini** [comment]
Ah ok. So you fixed a different but somewhat related issue #9728.

**Comment by neil-marcellini** [comment]
That proposal looks good to me too! The only tiny thing I would change is updating the condition to avoid negations. I think it's more clear like so.
`if (!this.state.isFocused || keyBoardEvent.key === 'ArrowUp' || keyBoardEvent.key === 'ArrowDown')`

@errahsoufiane Please put up a PR and we will give it a review.

**Comment by errahsoufiane** [comment]
@neil-marcellini applied job and created PR
@parasharrajat please check PR

**Comment by neil-marcellini** [comment]
The PR has been merged, what's the status on payment?

**Comment by errahsoufiane** [comment]
> The PR has been merged, what's the status on payment?

I am not hired yet, though I submitted proposal on upwork

**Comment by dylanexpensify** [comment]
We need applications to the job FYI!

**Comment by dylanexpensify** [comment]
@vladnyk @parasharrajat any updates here on applying for the job?

**Comment by dylanexpensify** [comment]
Sure can!

**Comment by dylanexpensify** [comment]
@parasharrajat sent!
@daraksha-dk mind applying for the reporting payment?

**Comment by dylanexpensify** [comment]
not overdue

**Comment by dylanexpensify** [comment]
@parasharrajat @daraksha-dk sent offers!
